hbase-server/src/main/java/org/apache/hadoop/hbase/HBaseServerBase.java

```diff
@@ -617,15 +616,45 @@ public ConfigurationManager getConfigurationManager() {
   public void updateConfiguration() {
     LOG.info("Reloading the configuration from disk.");
     // Reload the configuration from disk.
+    preUpdateConfiguration();
     conf.reloadConfiguration();
     configurationManager.notifyAllObservers(conf);
+    postUpdateConfiguration();
+  }
+
+  private void preUpdateConfiguration() {
+    try {
+      CoprocessorHost<?, ?> coprocessorHost = getCoprocessorHost();
+      if (coprocessorHost instanceof RegionServerCoprocessorHost) {
+        ((RegionServerCoprocessorHost) coprocessorHost).preUpdateConfiguration(conf);
+      } else if (coprocessorHost instanceof MasterCoprocessorHost) {
+        ((MasterCoprocessorHost) coprocessorHost).preUpdateConfiguration(conf);
+      }
+    } catch (IOException e) {
+      LOG.error("Error while calling coprocessor preUpdateConfiguration()", e);
```

Comment:
Instead of catching there here, can we throw them up the call chain? In particular, this will show up in client log when the reload request comes in via RPC. Maybe the signal handler will have to log.

Reply:
done